flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/MessageAcknowledgingSourceBase.java

```diff
@@ -177,21 +177,19 @@ protected boolean addId(UId uid) {
 	}
 
 	@Override
-	public void restoreState(SerializedCheckpointData[] state) throws Exception {
```

Comment:
Some how in my version of the code am not seeing this synchronization at all on pendingCheckpoints.